File: presenter/src/main/java/br/com/luisfelipeas5/givemedetails/presenter/di/PresenterModule.java
package br.com.luisfelipeas5.givemedetails.presenter.di;

import br.com.luisfelipeas5.givemedetails.model.datamangers.MovieMvpDataManager;
import br.com.luisfelipeas5.givemedetails.presenter.list.MoviesMvpPresenter;
import br.com.luisfelipeas5.givemedetails.presenter.list.MoviesPresenter;
import br.com.luisfelipeas5.givemedetails.presenter.schedulers.AppSchedulerProvider;
import br.com.luisfelipeas5.givemedetails.presenter.schedulers.SchedulerProvider;
import dagger.Module;
import dagger.Provides;

@Module
public class PresenterModule {

    @Provides
    SchedulerProvider provideSchedulerProvider() {
        return new AppSchedulerProvider();
    }

    @Provides
    MoviesMvpPresenter provideMoviesMvpPresenter(MovieMvpDataManager movieMvpDataManager,
                                                 SchedulerProvider schedulerProvider) {
        return new MoviesPresenter(movieMvpDataManager, schedulerProvider);
    }

}

File: app/src/main/java/br/com/luisfelipeas5/gimmedetails/di/modules/PresenterModule.kt
package br.com.luisfelipeas5.gimmedetails.di.modules

import br.com.luisfelipeas5.gimmedetails.ui.medias.MediasContract
import br.com.luisfelipeas5.gimmedetails.ui.medias.MediasPresenter
import dagger.Module
import dagger.Provides

@Module
class PresenterModule {

    @Provides
    fun provideMediasPresenter(): MediasContract.Presenter {
        return MediasPresenter()
    }

}